Virtual Librarian/VirtualLibrarian.API/Controllers/BooksController.cs

```diff
@@ -60,7 +59,7 @@ public IHttpActionResult GetAllBooks()
         [Route("GetBook/{id}")]
         public IHttpActionResult GetBook(int id)
         {
-            Book book = library.GetBook(id);
+            Book book = _library.GetBook(id);
             if (book != null)
```

Comment:
You should not check for `null`s in the controller. Throw an exception in `ILibrary` and handle it in controller

Reply:
Same in all other routes